daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala

```diff
@@ -1087,7 +1087,9 @@ private[lf] object DecodeV1 {
       proto: PLF.PrimType,
       bTyp: BuiltinType,
       minVersion: LV = LV.Features.default
-  )
+  ) {
+    val typ = TBuiltin(bTyp)
```

Comment:
If we don't always use `info.typ`, you could consider making it lazy to not allocate it when creating a value of `BuiltinTypeInfo`.

Reply:
There is a fix number of BuiltinTypeInfo allocated.